api/uniter/uniter.go

```diff
@@ -395,6 +395,51 @@ func (st *State) AllMachinePorts(machineTag names.MachineTag) (map[network.PortR
 	return portsMap, nil
 }
 
+// OpenedMachinePortRanges returns all port ranges currently open on the given
+// machine, grouped by unit tag and application endpoint.
+// relation that applies.
+func (st *State) OpenedMachinePortRanges(machineTag names.MachineTag) (map[names.UnitTag]map[string][]network.PortRange, error) {
+	if st.BestAPIVersion() < 17 {
+		// OpenedMachinePortRanges() was introduced in UniterAPIV17.
+		return nil, errors.NotImplementedf("AllMachinePorts() (need V17+)")
```

Comment:
Copy Paste error?  s/AllMachinePorts/OpenedMachinePortRanges/

Reply:
Nice catch; I will fix that